docs: add more warnings for default secrets and docker-compose (#27921)
apache/superset: docs/docs/installation/configuring-superset.mdx

You can generate a strong secure key with `openssl rand -base64 42`.

> > Your secret key will be used for securely signing session cookies

Reviewer: ```suggestion
:::caution Your secret key will be used for securely signing session cookies
```

>   and encrypting sensitive information on the database

Reviewer: ```suggestion
and encrypting sensitive information stored in Superset's application metadata database.
```

>   Make sure you are changing this key for your deployment with a strong key.

Reviewer: ```suggestion
Ensure your deployment uses a strong, unique key.
Store this key securely. Without this key, you will be unable to access your Superset metadata database. That is, you will lose all of your Superset dashboards, charts, etc. :::
```

Reviewer 2: This `:::caution` block as merged is not closed with a `:::`, so this page renders everything after like so:

![image](https://github.com/apache/superset/assets/46089/ca2ce153-78a0-450a-a87f-2fbbf713e317)

The suggestion did have a closing `:::`, though.

Reviewer: Thanks @csummers! I have a PR open now to fix this.
